src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -579,6 +609,15 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
                 out.writeVInt(INLINE_VAL); //Default to inline
             }
         }
+
+        @Override
+        public String toString() {
+            //ScriptModes looks for 'dynamic' rather than 'inline' when reading settings, we should rename but we want to keep bw comp for now.
+            if (this == INLINE) {
+                return "dynamic";
```

Comment:
oh boy that means that we should keep this distinction in master too, cause it's not just about bw comp. we prefer inline internally but we always called it dynamic when talking to the outside world... but I see your point, coming up with inline now among settings is confusing.